pkg/scheduling/volumeusage.go

```diff
@@ -132,7 +131,7 @@ func resolveDriver(ctx context.Context, kubeClient client.Client, volumeName str
 	}
 	if storageClassName != "" {
 		driverName, err := driverFromSC(ctx, kubeClient, storageClassName)
-		if err != nil {
+		if err != nil && !errors.IsNotFound(err) {
 			return "", err
```

Comment:
I'm assuming that we should probably log if we don't find the storageClassName but the storage class name is specified. Also, I'd think that we should exit out of this function as soon as we get the `NotFound` error but we should just return a `nil` error when we are getting this data as part of state tracking

Reply:
So the reason I'm not logging it is because there is a valid case for the storage class not existing but being specified. If you create a PV that specifies the same StorageClass as a PVC and all of the other requirements have been met, they can bind even if the StorageClass doesn't exist. I haven't been able to find solid documentation on **why** this is the case (it seems pretty counter-intuitive to me) but it is demonstrated in the kubernetes docs ([ref](https://kubernetes.io/docs/tasks/configure-pod-container/configure-persistent-volume-storage/)) and I've been able to verify that in my own cluster.

Reply:
I think the reason I didn't return `nil` immediately in this case is because I didn't think it followed the return style I had seen elsewhere in the project so I was trying to be consistent. Probably easier to talk offline about what I mean by this, but I'll give it a brief shot. I've seen quite a few examples of this:
```go
if err := someLogic(); err != nil {
  return err
}
return nil
```
rather than this:
```go
return someLogic()
```
I interpreted this as a way to make it simple to add more cases in the future, which I think waiting to return here does. I'm completely open to changing this to return earlier though.

Reply:
> If you create a PV that specifies the same StorageClass as a PVC and all of the other requirements have been met, they can bind even if the StorageClass doesn't exist

Let's discuss this one a bit more. I'm trying to understand what this looks like exactly and why someone would want to do this.

Reply:
Resolving, discussed and aligned offline. The referenced Kubernetes doc above in the thread is the best example I could find of a use case.